pennylane_lightning/core/src/simulators/lightning_qubit/measurements/MeasurementsLQubit.hpp

```diff
@@ -573,30 +574,38 @@ class Measurements final
      * Reference: https://en.wikipedia.org/wiki/Alias_method
      *
      * @param num_samples The number of samples to generate.
+     * @param catalyst_rng The seeded random number generator from catalyst.
      * @return 1-D vector of samples in binary, each sample is
      * separated by a stride equal to the number of qubits.
      */
-    std::vector<std::size_t> generate_samples(const std::size_t num_samples) {
+    std::vector<std::size_t> generate_samples(
+        const std::size_t num_samples,
+        const std::optional<std::mt19937> &catalyst_rng = std::nullopt) {
         const std::size_t num_qubits = this->_statevector.getNumQubits();
         std::vector<std::size_t> wires(num_qubits);
         std::iota(wires.begin(), wires.end(), 0);
-        return generate_samples(wires, num_samples);
+        return generate_samples(wires, num_samples, catalyst_rng);
     }
 
     /**
      * @brief Generate samples.
      *
      * @param wires Sample are generated for the specified wires.
      * @param num_samples The number of samples to generate.
+     * @param catalyst_rng The seeded random number generator from catalyst.
      * @return 1-D vector of samples in binary, each sample is
      * separated by a stride equal to the number of qubits.
      */
-    std::vector<std::size_t>
-    generate_samples(const std::vector<std::size_t> &wires,
-                     const std::size_t num_samples) {
+    std::vector<std::size_t> generate_samples(
+        const std::vector<std::size_t> &wires, const std::size_t num_samples,
+        const std::optional<std::mt19937> &catalyst_rng = std::nullopt) {
         const std::size_t n_wires = wires.size();
         std::vector<std::size_t> samples(num_samples * n_wires);
-        this->setRandomSeed();
+        if (catalyst_rng.has_value()) {
+            this->setRNG(catalyst_rng.value());
```

Comment:
Since we decided it was fine to seed the kokkos rng instance with a seed taken from the catalyst rng, would it not be simpler, and more independent (e.g. we definitely shouldn't call this variable "catalyst"_rng), to provide a seed to the sample function?
Something along the lines of:
```c++
std::vector<std::size_t> generate_samples(
        const std::vector<std::size_t> &wires, const std::size_t num_samples,
        const std::size_t seed = 0) {
    ...
    if (seed) {
        this->setSeed(seed);
    } else {
        this->setRandomSeed();
    }
```

This eliminates the need for the `setRNG` function entirely, and will be uniform for both qubit and kokkos.